pkg/beacon/relay/registry/groups_test.go

```diff
@@ -118,42 +117,72 @@ func TestLoadGroup(t *testing.T) {
 
 func TestUnregisterStaleGroups(t *testing.T) {
```

Comment:
Instead of incorporating those assertions into this test, I'd implement a separate test checking whether `IsStale` was NOT executed for the last group passed - this is what we actually want to test. I'd also write "first group", "second group", and "third group" in assertions as it's hard to figure out by group public key what's happening when assertion fails.

```
func TestUnregisterStaleGroupsSkipLastGroupCheck(t *testing.T) {
	mockChain := &mockGroupRegistrationInterface{
		groupsToRemove:       [][]byte{},
		groupsCheckedIfStale: make(map[string]bool),
	}

	gr := NewGroupRegistry(mockChain, persistenceMock)

	gr.RegisterGroup(signer1, channelName1)
	gr.RegisterGroup(signer2, channelName1)
	gr.RegisterGroup(signer3, channelName1)

	gr.unregisterStaleGroups(signer3.GroupPublicKeyBytes())

	group1PublicKeyString := groupKeyToString(signer1.GroupPublicKeyBytes())
	if mockChain.groupsCheckedIfStale[group1PublicKeyString] != true {
		t.Fatalf("IsStaleGroup() was expected to be called for the first group")
	}

	group2PublicKeyString := groupKeyToString(signer2.GroupPublicKeyBytes())
	if mockChain.groupsCheckedIfStale[group2PublicKeyString] != true {
		t.Fatalf("IsStaleGroup() was expected to be called for the second group")
	}

	group3PublicKeyString := groupKeyToString(signer3.GroupPublicKeyBytes())
	if mockChain.groupsCheckedIfStale[group3PublicKeyString] != false {
		t.Fatalf("IsStaleGroup() was expected to not be called for the third group")
	}
}
```

Reply:
Agree, the original test looked cluttered with these many validations. Created a new one.